coders/fits.c

```diff
@@ -145,27 +145,34 @@ static MagickBooleanType IsFITS(const unsigned char *magick,const size_t length)
 
 static inline double GetFITSPixel(Image *image,int bits_per_pixel)
 {
+  double pixel = 0;
   switch (image->depth >> 3)
   {
     case 1:
-      return((double) ReadBlobByte(image));
+      pixel=((double) ReadBlobByte(image));
```

Comment:
You are missing `break` statements.

Reply:
Updated, returns are kept now, so this doesn't apply